pkg/download/cache_fill_middleware.go

```diff
@@ -0,0 +1,62 @@
+package download
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/log"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/storage"
+)
+
+// CacheFillMiddleware ensures that next is called with the module@version in the cache.
+// Access the full module@version in your handler with c.Value("versionInfo").
+// ParamsMiddleware must be in the middleware chain before this middleware.
+func CacheFillMiddleware(
+	next buffalo.Handler,
+	fetcher module.Fetcher,
+	getterSaver storage.GetterSaver,
+	lggr *log.Logger,
+) buffalo.Handler {
+	const op errors.Op = "download.fetchMiddleware"
+	return func(c buffalo.Context) error {
+		module := c.Value("module").(string)
+		version := c.Value("version").(string)
+		versionInfo, err := getterSaver.Get(module, version)
+		if storage.IsNotFoundError(err) {
+			ref, err := fetcher.Fetch(module, version)
+			if err != nil {
+				err := errors.E(op, err)
+				lggr.SystemErr(err)
+				return err
+			}
+			defer ref.Clear()
+			version, err := ref.Read()
+			if err != nil {
+				err := errors.E(op, err)
+				lggr.SystemErr(err)
+				return err
+			}
+			defer version.Zip.Close()
+		} else if err != nil {
+			err := errors.E(op, err)
+			lggr.SystemErr(err)
+			return err
+		}
+		c.Set("versionInfo", version)
```

Comment:
Is this meant to set the `versionInfo` param to the string `version` from 23?

Reply:
this should set to `versionInfo`, thanks for catching this!